docker-compose.yml

```diff
@@ -22,6 +20,9 @@ services:
       MINIO_ACCESS_KEY: minioAdminAccessKey
       MINIO_SECRET_KEY: minioAdminSecretKey
     volumes:
-      - /data
-    ports:
-      - 9000:9000
+      - fsdata:/data
+
+volumes:
+  dbdata:
```

Comment:
Why the need for named volumes?

Reply:
If I do `docker-compose up`, it creates volumes with long hash-like names.  If I then do `docker-compose down` and `docker-compose up`, it does **not** reuse the old names.  Rather, it creates **new** hash-like names.  This makes the data functionally non-persistent.  Further, each time you do an up and down, you end up creating more volumes that use storage space that are never automatically cleaned up in any manner.  In addition to being functionally non-persistent, it is also resource consuming.  Using named volumes works around this.

If there is a better way, I would be happy to know it.

Reply:
`docker-compose stop` (not `down`) should be used to stop the containers. It persists and reuses the containers and volumes for the next `up`. For that reason, `stop` is the command suggested in the README.

Even if you have named volumes, if you want to cleanly destroy everything with `down`, you need [to pass the `-v` / `--volumes` flag](https://docs.docker.com/compose/reference/down/), which will remove both named and anonymous volumes.

Unless we need named volumes, I think it's a better practice to let Docker Compose manage volume names anonymously, which ensures there are never any name conflicts.

Reply:
Using `docker-compose down` should **NOT** destroy persistent data.  This is a terrible default behavior.  Using named volumes fixes this.

Sure, stopping and starting is nice, but there are some things that stopping does allow -- for instance if you want to change volume mounts (for instance, to bind mount in a directory to easily access additional code within the container).

Reply:
Following our conversation, I agree with your fundamental point.

I think we should prefix these volume names with `rgd-` to prevent conflicts with other projects which may create volumes with the same names.

Reply:
The names are already prefixed with "resonantgeodata_" by docker-compose.